python/Ganga/new_tests/GPI/newXMLTest/TestXMLCorruption.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import absolute_import
+
+from ..GangaUnitTest import GangaUnitTest
+
+from os import path, stat, unlink
+
+import time
+
+from .utilFunctions import getJobsPath, getXMLDir, getXMLFile, getIndexFile
+
+global_AutoStartReg = True
+
+notXMLStr = ['ThisIsNOTXML' for _ in range(20)]
+badStr = ''.join(notXMLStr)
+
+class TestXMLGenAndLoad(GangaUnitTest):
+
+    def setUp(self):
+        """Make sure that the Job object isn't destroyed between tests"""
+        super(TestXMLGenAndLoad, self).setUp()
+        from Ganga.Utility.Config import setConfigOption
+        setConfigOption('TestingFramework', 'AutoCleanup', 'False')
+        setConfigOption('Configuration', 'AutoStartReg', global_AutoStartReg)
+
+    def test_a_JobConstruction(self):
+        """ First construct the Job object (singular)"""
+        from Ganga.Utility.Config import getConfig
+        self.assertFalse(getConfig('TestingFramework')['AutoCleanup'])
+
+        from Ganga.GPI import Job, jobs
+        j=Job()
+        assert len(jobs) == 1
+
+        global global_AutoStartReg
+        global_AutoStartReg = False
+
+    def test_b_TestRemoveXML(self):
+
+        XMLFileName = getXMLFile(0)
+
+        unlink(XMLFileName)
+
+        assert not path.isfile(XMLFileName)
+        assert path.isfile(XMLFileName+'~')
+
+        global global_AutoStartReg
+        global_AutoStartReg = True
+
+    def test_c_TestBackupLoad(self):
+
+        from Ganga.GPI import jobs
+
+        assert len(jobs) == 1
+
+        ## trigger load
+        backend2 = jobs(0).backend
+
+        assert backend2 is not None
+
+        global global_AutoStartReg
+        global_AutoStartReg = False
+
+    def test_d_TestCorruptXML(self):
+
+        XMLFileName = getXMLFile(0)
+
+        unlink(XMLFileName)
+        assert not path.isfile(XMLFileName)
+
+        handler = open(XMLFileName, 'w')
+        handler.write(badStr)
+        handler.flush()
+        handler.close()
```

Comment:
The old unittest system had bigger problems. I discovered it bleeding files everywhere when mis-using contexts.It also abused logging redirection enough that there were problems with reliably printing string to the terminal.